aldryn_search/base.py

```diff
@@ -130,7 +138,30 @@ def get_description(self, obj):
         """
         return None
 
+    def check_for_app_config(self, obj):
```

Comment:
This check isn't only about Pages, it's about every object that has at least one app_config (and CMS Pages and Titles isn't in that group). If the object has a `app_config.namespace = 'foo'` and we does not have a `Page.application_namespace = 'foo'` we will generate incorrect urls when calling `<obj>.get_absolute_url()` in time we index it.

An example... 
* Let's say that for aldryn-newsblog we have following `NewsBlogConfig` namespaces: `foo` and `bar`.
* Let's say that we have a Page with `application_namespace = 'bar'` and `application_urls = 'NewsBlogApp'`
* Let's say that we have the `<Article 1>` with `app_config.namespace = 'foo'`.

When indexing we get `<Article 1>.get_absolute_url()` and because it's namespace is `foo` and we does not have a newsblog page with `application_namespace = 'foo'` it will return url for page with `application_namespace = 'bar'`. This happens because how Django handle reverse url resolution.

This behaviour will index objects with wrong urls (with lead users to Http 404 when access it in search results) and this implementation will avoid that by not indexing this objects.